src/sentry/api/endpoints/data_export.py

```diff
@@ -42,16 +42,28 @@ def post(self, request, organization):
         data = serializer.validated_data
 
         try:
-            # TODO(Leander): Prevent repeated requests for identical queries per organization, if one is in progress
-            data_export = ExportedData.objects.create(
+            existing_data_exports = ExportedData.objects.filter(
                 organization=organization,
                 user=request.user,
                 query_type=data["query_type"],
                 query_info=data["query_info"],
-            )
+                date_finished=None,
+            ).order_by("-date_added")
+            # If this user has sent a sent a request with the same payload and organization,
+            # we return them the latest one that is NOT complete (i.e. don't start another)
+            if len(existing_data_exports) > 0:
+                data_export = existing_data_exports[0]
+                status = 200
+            else:
+                data_export = ExportedData.objects.create(
```

Comment:
@leeandher You probably should use `get_or_create` instead of what you're doing. See: https://docs.djangoproject.com/en/3.0/ref/models/querysets/#get-or-create

Reply:
Note this comment should supersede the ones above so I will resolve those

Reply:
Didn't know about that, that's so convenient -- i'll check it out